app/controllers/katello/api/registry/registry_proxies_controller.rb

```diff
@@ -225,6 +325,21 @@ def start_upload_blob
       pulp_response.headers.each do |key, value|
         response.header[key.to_s] = value
       end
+      
+      # if pulp was successful and we need to create a new repository, create it now
+      if 200 <= pulp_response.code && pulp_response.code <= 299 && 
+        @product.root_repositories.where(label:@container_name).empty?
+        
+        root = @product.add_repo(
+          name: @container_name,
+          label: @container_name,
+          download_policy: 'immediate',
+          content_type: Repository::DOCKER_TYPE,
+          unprotected: true
+        )
+        sync_task(::Actions::Katello::Repository::CreateRoot, root, @container_path_input) 
```

Comment:
One issue with using CreateRoot -- it creates entities that are unnecessary like a Pulp repository and a distribution for that repository.  I think this might be breaking you code, since the version_href after the Katello repository gets created incorrectly points to an empty Pulp repository.

Reply:
This is now working properly after updating Repository::Create. Thank you for the help!